src/amo/components/LandingPage.js

```diff
@@ -41,9 +44,33 @@ export class LandingPageBase extends React.Component {
 
   static defaultProps = {
     apiAddonType: getApiAddonType,
+    contentForType: null,
   }
 
-  contentForType(visibleAddonType) {
+  componentWillMount() {
+    this.setViewContextType();
+  }
+
+  componentDidUpdate() {
+    this.setViewContextType();
+  }
+
+  setViewContextType() {
+    const { apiAddonType, dispatch, params } = this.props;
+
+    try {
+      const addonType = apiAddonType(params.visibleAddonType);
+      dispatch(setViewContext(addonType));
+    } catch (err) {
+      if (err instanceof AddonTypeNotFound) {
+        log.info('AddonTypeNotFound in setViewContextType()', err);
```

Comment:
The issue with that is this page is hit any time and URL that matches `/*/` is accessed. I suppose in production this shouldn't *really* happen because we serve 404s usually from `addons-server` though any `Link` component pointing to a bad URL would cause this to throw.

We already do exactly what I'm doing here elsewhere in the component and have for awhile, in the interest of not throwing erroneous 500 errors when we're really just hitting a 404. And if there is an invalid `addonType` this will at least log the error.

It should be noted that `apiAddonType` will throw so we'll never `dispatch` with an invalid `addonType`.

If you're 100% sure about throwing here I've added it as a commit but that's a change from what this component already does and I'm doing this here because `apiAddonType` throws when it's handed an invalid `addonType`, which only happens here because of the router.